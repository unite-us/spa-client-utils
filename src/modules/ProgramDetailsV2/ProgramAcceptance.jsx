import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from '@unite-us/ui';

const acceptanceInfo = {
  accept: {
    text: 'Accepting referrals',
    icon: <Icon icon="IconCheckCircle" color="#117E15" className="mr-quarter" />,
  },
  deny: {
    text: 'Not accepting referrals',
    icon: <Icon icon="IconTimesCircle" color="#CF2A2A" className="mr-quarter" />,
  },
  deactivate: {
    text: 'Deactivated',
    icon: <Icon icon="IconMinusCircle" size={13.7} color="#CF2A2A" className="mr-quarter" />,
  },
};

function ProgramAcceptance(props) {
  const {
    receiving_referrals,
    status,
    hide_referrals,
    is_active,
  } = props;

  if (hide_referrals && !is_active) {
    return (
      <div className="program-acceptance mt-half">
        {
          acceptanceInfo.deactivate.icon
        }
        {
          <p>{acceptanceInfo.deactivate.text}</p>
        }
      </div>
    );
  }
  return (
    <div className="program-acceptance mt-half">
      {
        receiving_referrals ?
          acceptanceInfo.accept.icon :
          acceptanceInfo.deny.icon
      }
      { receiving_referrals ?
        <p>{acceptanceInfo.accept.text}</p> :
        <p>{acceptanceInfo.deny.text} {status && `(${status})`}</p>
      }
    </div>
  );
}

ProgramAcceptance.propTypes = {
  receiving_referrals: PropTypes.bool.isRequired,
  status: PropTypes.string,
  hide_referrals: PropTypes.bool,
  is_active: PropTypes.bool,
};

ProgramAcceptance.defaultProps = {
  status: '',
  hide_referrals: false,
  is_active: false,
};

export default ProgramAcceptance;
