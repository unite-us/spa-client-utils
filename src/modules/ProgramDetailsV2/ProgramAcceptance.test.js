import React from 'react';
import renderer from 'react-test-renderer';
import ProgramAcceptance from './ProgramAcceptance';

describe('ProgramAcceptance', () => {
  it('renders receiving referral programs', () => {
    const props = {
      receiving_referrals: true,
      status: null,
    };

    expect(renderer.create(<ProgramAcceptance {...props} />)).toMatchSnapshot();
  });

  it('renders non receiving referral programs', () => {
    const props = {
      receiving_referrals: false,
      status: 'Overfilled',
    };

    expect(renderer.create(<ProgramAcceptance {...props} />)).toMatchSnapshot();
  });

  it('renders decativated programs', () => {
    const props = {
      receiving_referrals: false,
      status: 'Overfilled',
      is_active: false,
      hide_referrals: true,
    };

    expect(renderer.create(<ProgramAcceptance {...props} />)).toMatchSnapshot();
  });
});
