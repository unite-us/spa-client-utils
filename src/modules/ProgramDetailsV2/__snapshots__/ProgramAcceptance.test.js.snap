// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgramAcceptance renders decativated programs 1`] = `
<div
  className="program-acceptance mt-half"
>
  <svg
    className="ui-icon mr-quarter"
    height="13.7px"
    id=""
    style={
      Object {
        "display": "inline-block",
        "fill": "#CF2A2A",
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 20 20"
    width="13.7px"
  >
    <path
      d="M10,0 C4.47580645,0 0,4.47580645 0,10 C0,15.5241935 4.47580645,20 10,20 C15.5241935,20 20,15.5241935 20,10 C20,4.47580645 15.5241935,0 10,0 Z M4.67741935,11.6129032 C4.41129032,11.6129032 4.19354839,11.3951613 4.19354839,11.1290323 L4.19354839,8.87096774 C4.19354839,8.60483871 4.41129032,8.38709677 4.67741935,8.38709677 L15.3225806,8.38709677 C15.5887097,8.38709677 15.8064516,8.60483871 15.8064516,8.87096774 L15.8064516,11.1290323 C15.8064516,11.3951613 15.5887097,11.6129032 15.3225806,11.6129032 L4.67741935,11.6129032 Z"
    />
  </svg>
  <p>
    Deactivated
  </p>
</div>
`;

exports[`ProgramAcceptance renders non receiving referral programs 1`] = `
<div
  className="program-acceptance mt-half"
>
  <svg
    className="ui-icon mr-quarter"
    height="16px"
    id=""
    style={
      Object {
        "display": "inline-block",
        "fill": "#CF2A2A",
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 27 32"
    width="16px"
  >
    <path
      d="M20.518 20.036q0-0.464-0.339-0.804l-3.232-3.232 3.232-3.232q0.339-0.339 0.339-0.804 0-0.482-0.339-0.821l-1.607-1.607q-0.339-0.339-0.821-0.339-0.464 0-0.804 0.339l-3.232 3.232-3.232-3.232q-0.339-0.339-0.804-0.339-0.482 0-0.821 0.339l-1.607 1.607q-0.339 0.339-0.339 0.821 0 0.464 0.339 0.804l3.232 3.232-3.232 3.232q-0.339 0.339-0.339 0.804 0 0.482 0.339 0.821l1.607 1.607q0.339 0.339 0.821 0.339 0.464 0 0.804-0.339l3.232-3.232 3.232 3.232q0.339 0.339 0.804 0.339 0.482 0 0.821-0.339l1.607-1.607q0.339-0.339 0.339-0.821zM27.429 16q0 3.732-1.839 6.884t-4.991 4.991-6.884 1.839-6.884-1.839-4.991-4.991-1.839-6.884 1.839-6.884 4.991-4.991 6.884-1.839 6.884 1.839 4.991 4.991 1.839 6.884z"
    />
  </svg>
  <p>
    Not accepting referrals
     
    (Overfilled)
  </p>
</div>
`;

exports[`ProgramAcceptance renders receiving referral programs 1`] = `
<div
  className="program-acceptance mt-half"
>
  <svg
    className="ui-icon mr-quarter"
    height="16px"
    id=""
    style={
      Object {
        "display": "inline-block",
        "fill": "#117E15",
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 27 32"
    width="16px"
  >
    <path
      d="M22.929 13.107q0-0.5-0.321-0.821l-1.625-1.607q-0.339-0.339-0.804-0.339t-0.804 0.339l-7.286 7.268-4.036-4.036q-0.339-0.339-0.804-0.339t-0.804 0.339l-1.625 1.607q-0.321 0.321-0.321 0.821 0 0.482 0.321 0.804l6.464 6.464q0.339 0.339 0.804 0.339 0.482 0 0.821-0.339l9.696-9.696q0.321-0.321 0.321-0.804zM27.429 16q0 3.732-1.839 6.884t-4.991 4.991-6.884 1.839-6.884-1.839-4.991-4.991-1.839-6.884 1.839-6.884 4.991-4.991 6.884-1.839 6.884 1.839 4.991 4.991 1.839 6.884z"
    />
  </svg>
  <p>
    Accepting referrals
  </p>
</div>
`;
